### PR TITLE
Change to typescript project

### LINKED ISSUE #2: use ts?
ts definitions here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/chrome/index.d.ts

### PR DESCRIPTION
**Pull request by Xadoy**
Snowpack is used to build all scripts. Chrome now should load the output `build/` folder instead of the root.
Input scripts in `src/` directory:
```
├── background.ts
├── options
│   └── options.ts
├── overlay
│   └── overlay.ts
└── popup.ts
```
Static files in `public/` directory:
```
├── images
│   ├── get_started128.png
│   ├── get_started16.png
│   ├── get_started32.png
│   └── get_started48.png
├── manifest.json
├── options
│   └── options.html
├── overlay
│   └── overlay.html
└── popup.html
```
Output `build/` would look like:
```
├── background.js
├── images
│   ├── get_started128.png
│   ├── get_started16.png
│   ├── get_started32.png
│   └── get_started48.png
├── manifest.json
├── options
│   ├── options.html
│   └── options.js
├── overlay
│   ├── overlay.html
│   └── overlay.js
├── popup.html
└── popup.js
```
This pull request solves #2

### CODE REVIEW

**Comment by mysterytony** [comment]
tql

File: package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
```
Comment (mysterytony) [inline package.json:1]: this probably changed how can a developer get started on this project? can you also update the readme doc?

File: snowpack.config.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// Snowpack Configuration File
```
Comment (mysterytony) [inline snowpack.config.js:1]: can we use https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/chrome/index.d.ts?

File: src/background.ts [hunk 3]
```diff
@@ -16,22 +16,22 @@ chrome.storage.sync.get("color", (items) => {
...
+let tabs_in_countdown: Record<number, number> = {};
 
 chrome.storage.sync.get("blocked_hostnames", (host_suffixes) => {
   console.log("time locked urls", host_suffixes);
   // read into blocked_hostnames
   blocked_hostnames = host_suffixes.blocked_hostnames || [];
 });
 
-function is_hostname_blocked(url, tabId) {
+function is_hostname_blocked(url: string, tabId: number) {
```
Comment (mysterytony) [inline src/background.ts:34]: do we also annotate return type?

**Comment by mysterytony** [comment]
is it the best practice to separate js and HTML files into `src` and `public` folder? can we group overlay, option, popup etc (by functional areas)?

File: snowpack.config.js (hunk 2 above)
Comment (Xadoy) [reply snowpack.config.js:1]: This is already the dev dependency, check line 27 of package.json. It's installed by `npm install --save-dev @types/chrome`, refer to the [usage](https://github.com/DefinitelyTyped/DefinitelyTyped#what-are-declaration-files-and-how-do-i-get-them) of those declaration files

File: src/background.ts (hunk 3 above)
Comment (Xadoy) [reply src/background.ts:34]: For this function, it's safe to omit return type. TypeScript is smart enough to infer the return type, we only return true/false, so it's easy to know it's boolean.

**Comment by Xadoy** [comment]
> is it the best practice to separate js and HTML files into `src` and `public` folder? can we group overlay, option, popup etc (by functional areas)?

It's a common practice when we introduce build system. Files in `public` will not be processed, and output directly to `build` or `dist`. I am not sure if that's the best practice, but the modern frontend frameworks - vue ([official doc](https://cli.vuejs.org/guide/build-targets.html#app), [example](https://github.com/gothinkster/vue-realworld-example-app)), React ([cra doc](https://create-react-app.dev/docs/using-the-public-folder/), [cra example](https://github.com/react-cosmos/create-react-app-example)), and umi.js, etc - use this convention. Also I am referring these projects ([chrome-extension-typescript-starter](https://github.com/chibat/chrome-extension-typescript-starter), [chrome-extension-react-typescript-boilerplate](https://github.com/martellaj/chrome-extension-react-typescript-boilerplate)) when I was making this change. Normally the use of html is minimal, and most of codes are written in js, ui is generated by js document api. I think it's better to keep html as a main entrance, but its job is just to load the corresponding javascript file. What do you think?

File: package.json (hunk 1 above)
Comment (Xadoy) [reply package.json:1]: Done in [57fa2dd](https://github.com/mysterytony/timelocker-chrome/pull/9/commits/57fa2ddd3797aef5870cbd873b6e0f71f7e00f03)

**Comment by mysterytony** [comment]
nb, merging now

**Pull request merged**